occrp/views.py

```diff
@@ -169,12 +158,45 @@ def about():
     return render_template('about.html')
 
 
-def get_or_create(model, **kwargs):
-    instance = db.session.query(model).filter_by(**kwargs).first()
-    if instance:
-        return (instance, False)
-    else:
-        instance = model(**kwargs)
-        db.session.add(instance)
-        db.session.commit()
-        return (instance, True)
+def get_facets(**kwargs):
+    facets_query = '''
+        SELECT trim({entity_type}.{field}) as facet, count({entity_type}.id) as facet_count 
+        FROM story
+        JOIN events_stories ON story.id = events_stories.story_id 
+        JOIN event ON events_stories.event_id = event.id 
+        JOIN {join_table} ON event.id = {join_table}.event_id 
+        JOIN {entity_type} ON {join_table}.{entity_type}_id = {entity_type}.id 
+        WHERE story.id={story_id}
+        AND plainto_tsquery('english', '{query}') @@ to_tsvector(event.title || ' ' || event.description || ' ' || event.significance)
```

Comment:
Did we say we were going to make people and organizations searchable, too? Might be worth joining those in here and adding them to the search vector.